src/helpers/dataFormatter.ts

```diff
@@ -0,0 +1,139 @@
+import { getEmptyQuestion, getEmptyCategory } from './dataCreator';
+import { Category } from '../types';
+
+const quizDataSchema = {
+  _meta: {
+    dataKey: 'quizId',
+    respKey: 'QuizId',
+  },
+  quizId: 'QuizId',
+  name: 'Name',
+  isDraft: 'IsDraft',
+  numberOfQuestionsPerCategory: 'NumberOfQuestionsPerCategory',
+  categories: {
+    _meta: {
+      dataKey: 'categoryId',
+      respKey: 'CategoryId',
+    },
+    categoryId: 'CategoryId',
+    categoryName: 'CategoryName',
+    questions: {
+      _meta: {
+        dataKey: 'questionId',
+        respKey: 'QuestionId',
+      },
+      questionId: 'QuestionId',
+      points: 'Points',
+      text: 'Text',
+      options: {
+        _meta: {
+          dataKey: 'optionId',
+          respKey: 'OptionId',
+        },
+        optionId: 'OptionId',
+        text: 'OptionText',
+        isCorrect: 'IsCorrect',
+      },
+    },
+  },
+};
+const gameSchema = {
+  _meta: {
+    dataKey: 'gameId',
+    respKey: 'GameId',
+  },
+  gameId: 'GameId',
+  winnerTeamId: 'WinnerTeamId',
+  currentTeamId: 'CurrentTeamId',
+  timeLimit: 'TimeLimit',
+  selectionTimeLimit: 'SelectionTimeLimit',
+  isComplete: 'IsComplete',
+  teams: {
+    _meta: {
+      dataKey: 'teamId',
+      respKey: 'TeamId',
+    },
+    teamId: 'TeamId',
+    name: 'TeamName',
+    score: 'Score',
+    selectedOptions: {
+      _meta: {
+        dataKey: 'teamQuestionMapId',
+        respKey: 'TeamQuestionMapId',
+      },
+      teamQuestionMapId: 'TeamQuestionMapId',
+      selectedOptionId: 'SelectedOptionId',
+      questionId: 'QuestionId',
+    },
+  },
+  quiz: quizDataSchema,
+};
+
+export const createQuizData = (acc: any, quiz: any, schema: any) => {
+  const metaData = schema._meta;
+  const index = acc.findIndex((x: any) => x[metaData.dataKey] === quiz[metaData.respKey]);
+  const data: any = index < 0 ? {} : acc[index];
+
+  for (const key in schema) {
+    if (key !== '_meta') {
+      if (typeof schema[key] === 'object') {
+        data[key] = createQuizData(data[key] || [], quiz, schema[key]);
+      } else {
+        data[key] = quiz[schema[key]];
+      }
+    }
+  }
```

Comment:
What are we trying to do here?
What does this function do?
Can you add a JSDoc for this?

Reply:
added JSDoc for `formatQuizzesData` function